qiskit_nature/utils/linalg.py

```diff
@@ -179,3 +179,142 @@ def _swap_columns(matrix: np.ndarray, i: int, j: int) -> None:
+        - `arXiv:1711.02242`_
+        - `International Journal of Quantum Chemistry 12, 683 (1977)`_
+
+    Args:
+        mat: The matrix to decompose.
+        error_threshold: Threshold for allowed error in the decomposition.
+            The error is defined as the maximum absolute difference between
+            an element of the original tensor and the corresponding element of
+            the reconstructed tensor.
+        max_vecs: The maximum number of vectors to include in the decomposition.
+
+    Returns:
+        The Cholesky vectors ``v_i`` assembled into a 2-dimensional Numpy array
+        whose columns are the vectors.
+
+    .. _arXiv:1711.02242: https://arxiv.org/abs/1711.02242
+    .. _International Journal of Quantum Chemistry 12, 683 (1977): https://doi.org/10.1002/qua.560120408
+    """
+    dim, _ = mat.shape
+
+    if max_vecs is None:
+        max_vecs = dim
+
+    cholesky_vecs = np.zeros((dim, max_vecs + 1), dtype=mat.dtype)
+    errors = np.real(np.diagonal(mat).copy())
+    for index in range(max_vecs + 1):
+        max_error_index = np.argmax(errors)
+        max_error = errors[max_error_index]
+        if max_error < error_threshold:
+            break
+        cholesky_vecs[:, index] = mat[:, max_error_index]
+        if index:
+            cholesky_vecs[:, index] -= (
+                cholesky_vecs[:, 0:index] @ cholesky_vecs[max_error_index, 0:index].conj()
+            )
+        cholesky_vecs[:, index] /= np.sqrt(max_error)
+        errors -= np.abs(cholesky_vecs[:, index]) ** 2
+
+    return cholesky_vecs[:, :index]
+
+
+def double_factorized(
+    two_body_tensor: np.ndarray, *, error_threshold: float = 1e-8, max_vecs: int | None = None
+) -> tuple[np.ndarray, np.ndarray]:
+    r"""Double-factorized decomposition of a two-body tensor.
+
+    The double-factorized decomposition is a representation of a two-body tensor
+    :math:`h_{pqrs}` as
+
+    .. math::
+        h_{pqrs} = \sum_{t=1}^N \sum_{k\ell} U^{t}_{pk} U^{t}_{qk}
+            Z^{t}_{k\ell} U^{t}_{r\ell} U^{t}_{s\ell}
+
+    Here each :math:`Z^{(t)}` is a real symmetric matrix, referred to as a
+    "diagonal Coulomb matrix," and each :math:`U^{t}` is a unitary matrix, referred to
+    as an "orbital rotation."
+
+    The number of terms :math:`N` in the decomposition depends on the allowed
+    error threshold. A larger error threshold may yield a smaller number of terms.
+    Furthermore, the ``max_vecs`` parameter specifies an optional upper bound
+    on :math:`N`. The ``max_vecs`` parameter is always respected, so if it is
+    too small, then the error of the decomposition may exceed the specified
+    error threshold.
+
+    The input tensor is assumed to be positive definite when reshaped into a matrix.
+
+    .. warning::
+        No checking is performed to verify whether the input tensor is positive definite.
+        If the input tensor is not positive definite, then the decomposition returned will be invalid.
+
+    References:
```

Comment:
It might be nice to have the titles of the papers here so one has an idea without clicking over to it to find out what the ref is about. Same in the other file with a similar ref.

Reply:
done.